src/helpers.rs

```diff
@@ -224,6 +224,20 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         bug!("No field named {} in type {}", name, base.layout().ty);
     }
 
+    fn project_field_hasname<P: Projectable<'tcx, Provenance>>(
+        &self,
+        base: &P,
+        name: &str,
+    ) -> bool {
```

Comment:
This doesn't project anything, so the current name is confusing. What about `projectable_has_field`?